sklearn/tests/test_pipeline.py

```diff
@@ -863,6 +864,20 @@ def test_step_name_validation():
                                  [[1]], [1])
 
 
+def test_set_conditional_params():
+    estimator = Pipeline([
+        ('a', Pipeline([
+            ('b', DummyRegressor())
+        ]))
+    ])
+    params = {
```

Comment:
Should we use a `collections.OrderedDict` to make this test more future-proof?

Reply:
Shrug. I think it distorts the purpose of the test somewhat if we *only* test with an OrderedDict.

Reply:
I think this is a bit clearer (without some of the subtlety of python version-specific dict ordering):

```py
estimator.set_params(a__b__alpha=0.001, a__b=Lasso())
```